rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositorySpec.scala

```diff
@@ -0,0 +1,146 @@
+package coop.rchain.rspace.nextgenrspace.history
+
+import java.nio.charset.StandardCharsets
+
+import coop.rchain.rspace.{
+  Blake2b256Hash,
+  DeleteContinuations,
+  DeleteData,
+  DeleteJoins,
+  HotStoreAction,
+  InsertContinuations,
+  InsertData,
+  InsertJoins,
+  Serialize
+}
+import monix.eval.Task
+import org.scalatest.{FlatSpec, Matchers, OptionValues}
+
+import scala.concurrent.duration._
+import monix.execution.Scheduler.Implicits.global
+import coop.rchain.rspace.internal.{Datum, WaitingContinuation}
+import coop.rchain.rspace.nextgenrspace.history.TestData.randomBlake
+import coop.rchain.rspace.trace.{Consume, Produce}
+import scodec.Codec
+import scodec.bits.ByteVector
+
+import scala.collection.concurrent.TrieMap
+import scala.util.Random
+import cats.implicits._
+
+class HistoryRepositorySpec
+    extends FlatSpec
+    with Matchers
+    with OptionValues
+    with InMemoryHistoryRepositoryTestBase {
+
+  implicit object stringSerialize extends Serialize[String] {
+
+    def encode(a: String): ByteVector =
+      ByteVector.view(a.getBytes(StandardCharsets.UTF_8))
+
+    def decode(bytes: ByteVector): Either[Throwable, String] =
+      Right(new String(bytes.toArray, StandardCharsets.UTF_8))
+  }
+
+  implicit val codecString: Codec[String] = implicitly[Serialize[String]].toCodec
+  type TestHistoryRepository = HistoryRepository[Task, String, String, String, String]
+
+  "HistoryRepository" should "process insert one datum" in withEmptyRepository { repo =>
+    val testDatum = datum(1)
+    val data      = InsertData[String, String](testChannelDataPrefix, testDatum :: Nil)
+    for {
+      nextRepo <- repo.process(data :: Nil)
+      data     <- nextRepo.getData(testChannelDataPrefix)
+      fetched  = data.head
+      _        = fetched shouldBe testDatum
+    } yield ()
+  }
+
+  val testChannelDataPrefix          = "channel-data"
+  val testChannelJoinsPrefix         = "channel-joins"
+  val testChannelContinuationsPrefix = "channel-continuations"
+
+  it should "process insert and delete of thirty mixed elements" in withEmptyRepository { repo =>
+    val data  = (0 to 10).map(insertDatum).toList
+    val joins = (0 to 10).map(insertJoin).toList
+    val conts = (0 to 10)
+      .map(insertContinuation)
+      .toList
+    val elems: List[HotStoreAction] = Random.shuffle(data ++ joins ++ conts)
+
+    val dataDelete                           = data.map(d => DeleteData[String](d.channel))
+    val joinsDelete                          = joins.map(j => DeleteJoins[String](j.channel))
+    val contsDelete                          = conts.map(c => DeleteContinuations[String](c.channels))
+    val deleteElements: List[HotStoreAction] = dataDelete ++ joinsDelete ++ contsDelete
```

Comment:
`++` on `List`s :-(. Better use `Vector`

Reply:
you feel that it's important in tests? ok